### PR TITLE
Unique items teleporters

### PR DESCRIPTION
**Pull request by BombasterDS**
## About the PR
Adds two teleporters for BSO and NTR that allows them teleport their lockers if station don't have them.

## Why / Balance
Since we can't map them on every map

## Technical details
Lockers create unique item marker to prevent abuse with locker breaking.
UniqueEntityChecker search for all markers and delete teleporter entity if found with proper marker name.

## Media
https://discord.com/channels/1202734573247795300/1236697829171527841/1288306630307483648

:cl:
- add: BSO and NTR now can teleport their lockers if they not mapped!

## Release Notes

- **New Features**
	- Introduced components to restrict item usage to specific stations.
	- Added unique entity markers to prevent duplication of specific items on stations.
	- Enhanced interaction mechanics with new event handling for item usage attempts.

- **Bug Fixes**
	- Improved event handling for item usage attempts to prevent unauthorized actions.

- **Chores**
	- Updated metadata for teleporters to include licensing and resource details.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
could this also be done for nukies to spawn in syndicate bombs? that's the way it's done in ss13

File: Resources/Prototypes/_Goobstation/Entities/Objects/Devices/unique_teleporters.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    sprite: _Goobstation/Objects/Devices/unique_teleporters.rsi
+    state: standart
+  - type: Item
+    sprite: _Goobstation/Objects/Devices/unique_teleporters.rsi
+  - type: UseOnStationOnly
+  - type: SpawnItemsOnUse
+    sound: /Audio/Items/Mining/fultext_launch.ogg
+    items: 
+    - id: TrashBananaPeel # i need to fill this with something :\
+
```
[the remark this answers is not in the document]
Comment (BombasterDS) [reply Resources/Prototypes/_Goobstation/Entities/Objects/Devices/unique_teleporters.yml:18]: items field is required field, so i can left it empty

[the remark this answers is not in the document]
Comment (BombasterDS) [reply Resources/Prototypes/_Goobstation/Entities/Objects/Devices/unique_teleporters.yml]: I lied! It worked because it's abstract class :trollface:

**Comment by BombasterDS** [comment]
github not allows me to push commit (dead russia servers :()

image

**Review by Piras314 — CHANGES_REQUESTED** [review]
namespace is _Goobstation, other than that looks good though

**Comment by BombasterDS** [comment]
> namespace is _Goobstation, other than that looks good though

Fixed

**Review by Piras314 — APPROVED** [review]

**Review by whateverusername0 — APPROVED** [review]

File: Resources/Prototypes/_Goobstation/Entities/Objects/Devices/unique_teleporters.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aidenkrz) [reply Resources/Prototypes/_Goobstation/Entities/Objects/Devices/unique_teleporters.yml]: yes remove that learning

**Pull request merged**